### DISCUSSION TITLE
Difference between Actor and ActorRef

### DISCUSSION THREAD

**Discussion by alexamy** [statelyai/xstate#4909] [Q&A] [answered]
I am a little puzzled with difference between these two. Especially looking into `@xstate/solid` [useActorRef](https://github.com/statelyai/xstate/blob/main/packages/xstate-solid/src/useActorRef.ts), simplified version below:
```ts
import { createActor } from 'xstate';

function useActorRef(...): ActorRef {
  const actorRef = createActor(...); // type is `Actor`
  return actorRef as any; // type becomes `ActorRef`
}
```

What are the differences, and why can’t they be used interchangeably?

**Comment by davidkpiano** [comment, marked as answer]
For most intents and purposes, the two should be able to be used interchangeably. In short, `Actor` implements `ActorRef` but has a few extra methods.

Can you share code where this is not working as expected?

**Reply by alexamy** [reply]
I am using XState 5.13.0. This code has type errors for both `Actor` and `ActorRefFrom` types from `xstate`:
```ts
import { ActorRefFrom, Actor, createActor, setup } from 'xstate';

const child = setup({}).createMachine({
  id: 'child machine',
});

const parent = setup({
  types: {
    input: {} as { children: ActorRefFrom<typeof child> }, // <-- Actor<typeof child> is not working too
    context: {} as { children: ActorRefFrom<typeof child> }, // <-- Actor<typeof child> is not working too
  },
}).createMachine({
  id: 'parent machine',
  context: ({ input }) => ({
    children: input.children,
  }),
});

const childActor = createActor(child);
const parentActor = createActor(parent, {
  input: {
    children: [childActor], // <-- Type error here: 
    // Type 'Actor' is missing the following properties from type 'ActorRef'
  },
});
```

The `parent` machine purpose is to control actors that have already been spawned by the external code.

**Reply by davidkpiano** [reply]
Why is it in an array? There is no type error if you move it out of the array:

```diff
const parentActor = createActor(parent, {
  input: {
-   children: [childActor],
+   children: childActor,
  },
});
```

**Reply by alexamy** [reply]
Thanks, my bad.
